genrl/deep/agents/ppo1/ppo1.py

```diff
@@ -221,71 +187,12 @@ def update_policy(self):
             torch.nn.utils.clip_grad_norm_(self.value_fn.parameters(), 0.5)
             self.optimizer_value.step()
 
-    def collect_rollouts(self, state):
-
-        for i in range(2048):
-
-            with torch.no_grad():
-                action, values, old_log_probs = self.select_action(state)
-
-            next_state, reward, dones, _ = self.env.step(np.array(action))
-            self.epoch_reward += reward
-
-            if self.render:
-                self.env.render()
-
-            self.rollout.add(
-                state,
-                action.reshape(self.env.n_envs, 1),
-                reward,
-                dones,
-                values,
-                old_log_probs,
-            )
-
-            state = next_state
-
-            for i, done in enumerate(dones):
-                if done:
-                    self.rewards.append(self.epoch_reward[i])
-                    self.epoch_reward[i] = 0
-
-        return values, dones
-
-    def learn(self):  # pragma: no cover
-        # training loop
-        state = self.env.reset()
-        for epoch in range(self.epochs):
-            self.epoch_reward = np.zeros(self.env.n_envs)
-
-            self.rollout.reset()
-            self.rewards = []
-
-            values, done = self.collect_rollouts(state)
-
-            self.get_traj_loss(values, done)
-
-            self.update_policy()
-
-            if epoch % 1 == 0:
-                print("Episode: {}, reward: {}".format(epoch, np.mean(self.rewards)))
-                self.rewards = []
-
-            if self.save_model is not None:
-                if epoch % self.save_interval == 0:
-                    self.checkpoint = self.get_hyperparams()
-                    self.save(self, epoch)
-                    print("Saved current model")
-
-        self.env.close()
-
     def get_hyperparams(self) -> Dict[str, Any]:
         hyperparams = {
             "network_type": self.network_type,
-            "timesteps_per_actorbatch": self.timesteps_per_actorbatch,
+            "batch_size": self.batch_size,
             "gamma": self.gamma,
             "clip_param": self.clip_param,
-            "actor_batch_size": self.actor_batch_size,
             "lr_policy": self.lr_policy,
             "lr_value": self.lr_value,
             "policy_weights": self.policy_new.state_dict(),
```

Comment:
Save `rollout_size` here too. Atari envs will need a different rollout size so needs to be saved separately.

Reply:
cool